NitroxClient/MonoBehaviours/MultiplayerMovementController.cs

```diff
@@ -0,0 +1,174 @@
+using System;
+using System.Collections.Generic;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic;
+using NitroxModel.Core;
+using NitroxModel.DataStructures;
+using NitroxModel.Packets;
+using NitroxModel_Subnautica.DataStructures;
+using UnityEngine;
+
+namespace NitroxClient.MonoBehaviours;
+
+public class MultiplayerMovementController : MonoBehaviour
+{
+    private static readonly Dictionary<NitroxId, MultiplayerMovementController> movementControllersById = new();
+
+    public float TimeScalar { get; set; } = 1f;
+    public Vector3 TargetPosition { get; set; }
+    public Quaternion TargetRotation { get; set; }
+    public bool Receiving { get; private set; }
+    public bool Broadcasting { get; private set; }
+    public Vector3 Velocity { get; private set; }
+
+    public event Action BeforeUpdate = () => { };
+    public event Action BeforeFixedUpdate = () => { };
+    public event Action AfterUpdate = () => { };
+    public event Action AfterFixedUpdate = () => { };
```

Comment:
Only AfterUpdate is used here. Are the other events that could be used in a reasonably soon future?

Reply:
I added them to provide some extensibility to the MovementController, figured since I had AfterUpdate i might as well provide the rest, FixedUpdate I can potentially see getting some usage if I'm being honest, the BeforeUpdate method has some potential use cases, granted not many

Reply:
But is it more a, can have or will have? After all this is not to difficult to write again

Reply:
I have no immediate plans to put it to use, was simply something for future Nitrox or Mod usage.